bokehjs/src/coffee/models/annotations/arrow.coffee

```diff
@@ -15,13 +15,14 @@ export class ArrowView extends AnnotationView
   connect_signals: () ->
     super()
     @connect(@model.change, () => @plot_view.request_render())
-    @connect(@model.source.change, () ->
-      @set_data(@model.source)
-      @plot_view.request_render())
+    @connect(@model.source.streaming, () -> @set_data(@model.source))
+    @connect(@model.source.patching, () -> @set_data(@model.source))
+    @connect(@model.source.change, () -> @set_data(@model.source))
 
   set_data: (source) ->
     super(source)
     @visuals.warm_cache(source)
+    @plot_view.request_render()
```

Comment:
since `set_data` is called in the arrow `initialize` statement, is the arrow rendered twice on the page load? (and same with the other annotations?)